### PR TITLE
add datadog formatters

### PR DESCRIPTION
**Pull request by btkostner**
This adds a `DatadogLogger` and a `Plug` `DatadogLogger`. Copied mostly from the Google version. Field names are from the [Datadog docs](https://docs.datadoghq.com/logs/processing/attributes_naming_convention/#default-standard-attribute-list) and should work out of the box for most people.

Let me know if there are things I missed, or anything you want changed.

Outputs:
```json
{
  "domain": ["elixir"],
  "logger": {
    "thread_name": "#PID<0.510.0>",
    "method_name": "Elixir.Appsignal.initialize/0"
  },
  "message": "AppSignal disabled.",
  "syslog": {
    "hostname": [10, 10, 100, 100, 100, 100, 100],
    "severity": "info",
    "timestamp": "2020-12-14T19:15:28.207Z"
  }
}
```

```json
{
  "domain": ["elixir"],
  "logger": {
    "thread_name": "#PID<0.712.0>",
    "method_name": "Elixir.LoggerJSON.Ecto.telemetry_logging_handler/4"
  },
  "message": "CREATE TABLE IF NOT EXISTS `schema_migrations` (`version` bigint, `inserted_at` datetime, PRIMARY KEY (`version`)) ENGINE = INNODB",
  "query": {
    "decode_time_μs": 1,
    "execution_time_μs": 164,
    "latency_μs": 475,
    "queue_time_μs": 309,
    "repo": "Recognizer.Repo"
  },
  "syslog": {
    "hostname": [10, 10, 100, 100, 100, 100, 100],
    "severity": "debug",
    "timestamp": "2020-12-14T19:16:47.782Z"
  }
}
```

```json
{
  "domain": ["elixir"],
  "duration": 3863403,
  "http": {
    "url": "http://localhost/create-account",
    "status_code": 200,
    "method": "GET",
    "referer": "http://localhost:4000/login",
    "request_id": "http_FlDCOItxeudZJ20AAADD",
    "useragent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.66 Safari/537.36",
    "url_details": {
      "host": "localhost",
      "port": 4000,
      "path": "/create-account",
      "queryString": "",
      "scheme": "http"
    }
  },
  "logger": {
    "thread_name": "#PID<0.1042.0>",
    "method_name": "Elixir.LoggerJSON.Plug.call/2"
  },
  "message": "",
  "network": {
    "client": {
      "ip": "127.0.0.1"
    }
  },
  "phoenix": {
    "controller": "Elixir.RecognizerWeb.Accounts.UserRegistrationController",
    "action": "new"
  },
  "request_id": "http_FlDCOItxeudZJ20AAADD",
  "syslog": {
    "hostname": [10, 10, 100, 100, 100, 100, 100],
    "severity": "info",
    "timestamp": "2020-12-14T19:16:55.088Z"
  }
}
```

### CODE REVIEW

File: lib/logger_json/plug/metadata_formatters/datadog_logger.ex [hunk 1]
```diff
@@ -0,0 +1,75 @@
+if Code.ensure_loaded?(Plug) do
+  defmodule LoggerJSON.Plug.MetadataFormatters.DatadogLogger do
+    @moduledoc """
+    This formatter builds a metadata which is natively supported by Datadog:
+
+      * `http` - see [LogEntry#HttpRequest](https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry#HttpRequest);
```
Comment (AndrewDryga) [inline lib/logger_json/plug/metadata_formatters/datadog_logger.ex:6]: I think the docs are off here

File: lib/logger_json/formatters/datadog_logger.ex [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  defp method_name(metadata) do
+    function = Keyword.get(metadata, :function)
+    module = Keyword.get(metadata, :module)
+
+    FormatterUtils.format_function(module, function)
+  end
+
+  defp node_hostname do
+    {:ok, hostname} = :inet.gethostname()
+    hostname
```
Comment (AndrewDryga) [inline lib/logger_json/formatters/datadog_logger.ex:54]: This returns a hostname as chartlist which is later encoded to list of integers (check your examples in PR comment). We need to cast the charlist to binary explicitly

File: lib/logger_json/formatters/datadog_logger.ex [hunk 3]
```diff
@@ -0,0 +1,56 @@
+defmodule LoggerJSON.Formatters.DatadogLogger do
+  @moduledoc """
+  DataDog formatter.
```
Comment (AndrewDryga) [inline lib/logger_json/formatters/datadog_logger.ex:3]: Please add a note that documentation for the format is here - https://docs.datadoghq.com/logs/processing/attributes_naming_convention/#default-standard-attribute-list

**Comment by AndrewDryga** [comment]
Hey @btkostner, sorry for taking so long to reply. This generally looks good to me but needs to clean up some docs from GKE. I notice there is a duplication issue arising and I had ideas how we can restructure this package into something behavior-based to avoid it, but that can be done later, I don't want to overwhelm you.

Did you have a chance to try this formatter in production?

**Comment by btkostner** [comment]
I tried this formatter a bit while I was developing it, but I haven't put it in a production environment yet. I can pull a screenshot of it in Datadog after the PR fixes are done if you want.

**Comment by AndrewDryga** [comment]
@btkostner Would it be used in production? I want to make sure it would have some real usage if we add it, because we will need to maintain it. Otherwise, we can add a notice to the docs that this formatted is experimental and I'll remove that notice once we have somebody who run it for some time in the real app.

**Comment by btkostner** [comment]
Yes. This will be used for System76 elixir projects in production. Along that line, what are your thoughts about other json formatters similar to the ecto one?

**Review by AndrewDryga — APPROVED** [review]

**Comment by AndrewDryga** [comment]
@btkostner from your examples they look good to me, if we need we can tune them later but until more people start to use this formatter :).

**Pull request merged**